api/api.yaml

```diff
@@ -2835,6 +2848,75 @@ components:
           description: The time the connection was last updated.
           format: date-time
           example: 2023-07-13T21:34:44.816354Z
+        authScheme:
+          type: string
+          description: The authentication scheme used for this connection.
+          example: oauth2
+          enum:
+            [
+              none,
+              apiKey,
+              basic,
+              oauth2/authorizationCode,
+              oauth2/authorizationCodePKCE,
+              oauth2/clientCredentials,
+              oauth2/password,
+              jwt,
+            ]
+        scopes:
+          type: array
+          description: The OAuth scopes for this connection.
+          items:
+            type: string
+        status:
+          type: string
+          description: The status of the connection.
+          example: active
+          enum: [created, working, bad_credentials]
+        oauth2AuthorizationCode:
+          $ref: "#/components/schemas/Oauth2AuthorizationCode"
+    Oauth2AuthorizationCode:
```

Comment:
can you modify GenerateConnection to use this too?

Reply:
Also I thought you said GetConnection currently doesn't return expiredAt or issuedAt?

Reply:
For now, it is not returning, so I didn't add it to `required` list. I decided not to add this in server repo because it was going to take some time and effort to include those time stamps and wanted to not include for the time being. But since it is not really important fields, and it might be confusing, I guess we should add when we decide to add those fields in. Will remove from this PR.